ttnn/cpp/ttnn/tensor/types.hpp

```diff
@@ -609,6 +632,15 @@ struct MultiDeviceHostStorage {
             std::lock_guard<std::mutex> lock(buffer_mtx);
             return buffers.find(device_id) != buffers.end();
         }
+
+        inline bool is_allocated() const {
+            std::lock_guard<std::mutex> lock(buffer_mtx);
+
+            return std::all_of(ordered_device_ids.begin(), ordered_device_ids.end(), [&buffers = this->buffers](auto&& device_id) {
```

Comment:
```suggestion
            return std::all_of(ordered_device_ids.begin(), ordered_device_ids.end(), [this](auto&& device_id) {
```
will allow you to use `buffers` as well. The `this` pointer is cheap to copy by value, and is the same amount of indirection as a reference to `buffers`.

Reply:
I am passing just one thing what I use from this class.
I'll consider your change and maybe use it. Not sure right now.